src/utils/core.ts

```diff
@@ -814,3 +814,11 @@ function removePrefix(token: string): string {
 	});
 	return token;
 }
+
+/**
+ * Detects whether a BOLT-11 Lightning invoice encodes a non-zero amount in the Human-Readable Part
+ * (HRP).
+ */
+export function invoiceHasAmountInHRP(invoice: string): boolean {
+	return /^ln[a-z]{2,}[1-9][0-9]*(?:[mun]|0p)?1/.test(invoice);
```

Comment:
```suggestion
	return /^ln[a-z]{2,}[1-9][0-9]*(?:[mun]|0p)?1/i.test(invoice);
```

As per other comment - let's fix the regex by making it case insensitive to support uppercase characters as per BOLT-11

Reply:
worked on both requests.